auth-api/src/auth_api/resources/v1/entity.py

```diff
@@ -128,6 +128,25 @@ def delete_entity(business_identifier):
     return response, status
 
 
+@bp.route('/<string:business_identifier>/authentication', methods=['GET', 'OPTIONS'])
+@cross_origin(origins='*')
+@TRACER.trace()
+@_jwt.requires_auth
+def get_entity_authentication(business_identifier):
+    """Get passcode or password for the Entity identified by the provided business identifier."""
+    # This route allows public users to see if businesses have a form of authentication.
+    # It's used by the business dashboard for magic link.
+    if ((entity := EntityService.find_by_business_identifier(business_identifier, skip_auth=True)) and
+    (contact := entity.get_contact())):
+        has_valid_pass_code = (entity.pass_code_claimed == 'f' and entity.pass_code is not None) or entity.corp_type_code in ['SP','GP']
+        return {
+            'contactEmail': contact.email,
```

Comment:
We should probably mask the email that comes back like we do everywhere else. There is an existing utility function for this.

`def mask_email(email: str) -> str:` (src/auth_api/utils/util.py)

@seeker25 what do you think?

Reply:
Yup, I may have missed this part

Reply:
Need new service property in entity:
```
    @property
    def pass_code_claimed(self):
        """Return the pass_code_claimed for this entity."""
        return self._model.pass_code_claimed
```

rename `entity.corp_type_code` -> `entity.corp_type`